server/app/actions/BlockDisabledProgramAction.java

```diff
@@ -0,0 +1,52 @@
+package actions;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import controllers.routes;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.DisplayMode;
+import play.mvc.Action;
+import play.mvc.Http.Request;
+import play.mvc.Result;
+import services.program.ProgramDefinition;
+import services.program.ProgramService;
+
+/**
+ * Action that ensures that program the user is trying to access is not disabled.
+ *
+ * <p>The action will redirect the request to the home page if the the program is disabled.
+ *
+ * <p>
+ */
+public class BlockDisabledProgramAction extends Action.Simple {
```

Comment:
(question) `BlockDisabledProgramAction` make it sound like it's checking if a specific program block (screen) is disabled. Would `ProgramDisabledAction` work better?

Reply:
Mmm I can see the confusion. Name changed! Thanks for the suggestion